src/wallet/rpcwallet.cpp

```diff
@@ -3463,57 +3476,112 @@ UniValue z_listreceivedbyaddress(const UniValue& params, bool fHelp)
     }
 
     UniValue result(UniValue::VARR);
-    std::vector<SproutNoteEntry> sproutEntries;
-    std::vector<SaplingNoteEntry> saplingEntries;
-    pwalletMain->GetFilteredNotes(sproutEntries, saplingEntries, fromaddress, nMinDepth, false, false);
 
     std::set<std::pair<PaymentAddress, uint256>> nullifierSet;
     auto hasSpendingKey = boost::apply_visitor(HaveSpendingKeyForPaymentAddress(pwalletMain), zaddr);
     if (hasSpendingKey) {
         nullifierSet = pwalletMain->GetNullifiersForAddresses({zaddr});
     }
 
+    NotesFilters filters;
+    filters.minDepth = nMinDepth;
+    filters.ignoreSpent = false;
+    filters.requireSpendingKey = false;
+    filters.cwallet = pwalletMain;
+
     if (boost::get<libzcash::SproutPaymentAddress>(&zaddr) != nullptr) {
-        for (SproutNoteEntry & entry : sproutEntries) {
+        const auto pa = *boost::get<libzcash::SproutPaymentAddress>(&zaddr);
+        const auto itr_sprout = pwalletMain->GetNotesByType(pa.GetHash(), NoteType::sprout, timestamp);
+        auto itr = itr_sprout.begin();
+
+        while (itr != itr_sprout.end() && result.size() < limit) {
```

Comment:
```suggestion
        for (auto it = it_sprout.begin(); it != it_sprout.end() && result.size() < limit; it++) {
```
and then remove the iterator increments below.

Reply:
https://github.com/zcash/zcash/pull/4399/commits/3138989d2467ac57935bbc14ce0641c6d784f0f2